src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -592,7 +592,16 @@ private async Task<bool> ScanTargetsAsync(IEnumerable<Skimmer<TContext>> skimmer
                     {
                         context = _fileContexts[item];
 
-                        DetermineApplicabilityAndAnalyze(context, skimmers, disabledSkimmers);
+                        rootContext.Logger.AnalyzingTarget(context);
+
+                        if (options.Traces.Contains(nameof(DefaultTraces.MemoryUsage), StringComparer.OrdinalIgnoreCase))
+                        {
+                            rootContext.Logger.LogMemoryUsage(context);
+                        }
+
+                        DetermineApplicabilityAndAnalyze(options, context, skimmers, disabledSkimmers);
+
+                        rootContext.Logger.TargetAnalysisCompleted(context);
```

Comment:
>rootContext.Logger.TargetAnalysisCompleted(context); [](http://example.com/codeflow?start=24&length=52)

when the analysis throws exception do we need the memory usage information? Should we move this call to finally block?

Reply:
moved to finally block.